eva/catalog/catalog_manager.py

```diff
@@ -322,191 +384,90 @@ def udf_io(
+            name (str):  name of the table catalog entry
+            format_type (FileFormatType): media type
 
-        Arguments:
-           udf_name: udf name to be dropped.
+        Raises:
+            CatalogError: if format_type is not supported
 
         Returns:
-           True if successfully deleted else False
+            TableCatalog: newly inserted table catalog entry
         """
-        return self._udf_service.drop_udf_by_name(udf_name)
-
-    def rename_table(self, curr_table: DataFrameMetadata, new_name: TableInfo):
-        return self._dataset_service.rename_dataset(curr_table, new_name.table_name)
-
-    def check_table_exists(self, database_name: str, table_name: str):
-        metadata = self._dataset_service.dataset_object_by_name(
-            database_name, table_name
-        )
-        if metadata is None:
-            return False
+        if format_type is FileFormatType.VIDEO:
+            columns = get_video_table_column_definitions()
+            table_type = TableType.VIDEO_DATA
+        elif format_type is FileFormatType.IMAGE:
+            columns = get_image_table_column_definitions()
+            table_type = TableType.IMAGE_DATA
         else:
-            return True
-
-    def get_all_udf_entries(self):
-        return self._udf_service.get_all_udfs()
+            raise CatalogError(f"Format Type {format_type} is not supported")
 
-    def get_all_table_entries(self):
-        return self._dataset_service.get_all_datasets()
+        return self.create_and_insert_table_catalog_entry(
+            TableInfo(name), columns, table_type=table_type
+        )
 
-    def get_media_metainfo_table(
-        self, input_table: DataFrameMetadata
-    ) -> DataFrameMetadata:
-        """Get a media metainfo table.
+    def get_media_metadata_table_catalog_entry(
+        self, input_table: TableCatalog
+    ) -> TableCatalog:
+        """Get table catalog entry for multimedia metadata table.
         Raise if it does not exists
         Args:
-            input_table (DataFrameMetadata): input media table
+            input_table (TableCatalog): input media table
 
         Returns:
-            DataFrameMetadata: metainfo table maintained by the system
+            TableCatalog: metainfo table entry which is maintained by the system
         """
         # use file_url as the metadata table name
         media_metadata_name = Path(input_table.file_url).stem
-        obj = self.get_dataset_metadata(None, media_metadata_name)
+        obj = self.get_table_catalog_entry(None, media_metadata_name)
         if not obj:
             err = f"Table with name {media_metadata_name} does not exist in catalog"
             logger.exception(err)
             raise CatalogError(err)
 
         return obj
 
-    def create_media_metainfo_table(
-        self, input_table: DataFrameMetadata
-    ) -> DataFrameMetadata:
-        """Create a media metainfo table.
+    def create_media_metadata_table_catalog_entry(
```

Comment:
This input parameter is the output from `create_multimedia_table_catalog_entry` function?

Reply:
Yes. For every media table, we maintain a structured metadata table with a list of filenames.

Reply:
I should add a readme with these details.

Reply:
Please feel free to suggest better function names. Thanks!

Reply:
What about `create_multimedia_table_catalog_entry` and `create_multimedia_metadata_table_catalog_entry`?

Also, should the function name be `create_` or `insert_` or `create_and_insert_`? Just want to keep consistent with `insert_table_catalog_entry` name.